site/source/tutorials/007_theming/index.md

```diff
@@ -0,0 +1,149 @@
+
+In order to theme our widgets, we must ensure that they each extend `ThemeableMixin` and should change their top level class name to `root`. `ThemeableMixin` provides a `this.classes` function for looking up class names in the provided `theme` file and applies the processed class names. We change the top level class names to `root` in order to provide a predictable way to target the outer-node of a widget.
+
+{% instruction 'Add the following to `Banner.ts`.' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/widgets/Banner.ts' lines:3-13 %}
+
+{% aside 'Reminder' %}
+If you cannot see the application, remember to run `dojo build -w` to build the application and start the development server.
+{% endaside %}
+
+This `Banner` widget will now have access to the classes in `banner.m.css` and can receive a `theme`. We use the `root` class to ensure that the theme we create can easily target the correct node.
+
+{% instruction 'Create a new CSS file for `Banner` named `banner.m.css`.' %}
+
+We will create an empty `root` class for now as our base theme does not require any styles to be added to the `Banner` widget.
+
+{% include_codefile 'demo/finished/biz-e-corp/src/styles/banner.m.css' lang:css %}
+
+{% instruction 'Now, let's look at changing `WorkerForm`' %}
+
+{% aside 'Fixed Classes' %}
+Fixed classes apply styles that _cannot_ be overridden by a theme, using a suffix is a convention that helps easily differentiate the intention of these classes.
+{% endaside %}
+
+`WorkerForm` already uses the `ThemeableMixin` and has a `workerForm` class on its root node. Let's change the workerForm class to a `root` class, and while we are there, we will create a `rootFixed` class too, and apply that as a `fixed` class to the root node. This passes a class name to the theming system that cannot be changed or overridden via a theme and ensures that structured or nested styles are not lost when a theme is used.
+
+{% include_codefile 'demo/finished/biz-e-corp/src/widgets/WorkerForm.ts' lines:48-51 %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/styles/workerForm.m.css' lines:1-17 lang:css %}
+
+If you open the application in a browser its appearance and behavior should be unchanged.
+
+{% instruction 'Now add a `root` and `rootFixed` to both `Worker.ts` and `WorkerContainer.ts`' %}
+
+Next, we will start to create a theme.
+
+{% section %}
+
+## Creating a Theme
+
+{% task 'Create a theme directory' %}
+
+Let's create a `theme` directory under your project `src` to store our theme resources, and create a `dojo` theme with `theme.ts` under `src/themes/dojo/theme.ts`. This is the `theme` resource where we will import our themed CSS files and export the keyed theme for consumption by our widgets.
```

Comment:
this line says to create a `theme` directory, but it is refered to as `themes` throughout. Need to pick one name.

Reply:
changed to `themes` as this is the directory name